test/javascripts/controllers/assign-user-test.js

```diff
@@ -0,0 +1,61 @@
+import { discourseModule } from "discourse/tests/helpers/qunit-helpers";
+import { test } from "qunit";
+import pretender from "discourse/tests/helpers/create-pretender";
+import EmberObject from "@ember/object";
+
+discourseModule("Unit | Controller | assign-user", function (hooks) {
+  hooks.beforeEach(function () {});
+
+  test("doesn't set suggestions if controller is destroyed", function (assert) {
+    pretender.get("/assign/suggestions", () => [
+      200,
+      { "Content-Type": "application/json" },
+      {
+        suggestions: [],
+        assign_allowed_on_groups: ["nat"],
+        assign_allowed_for_groups: [],
+      },
+    ]);
+    const controller = this.getController("assign-user", {
+      model: {
+        target: EmberObject.create({}),
+      },
+    });
+    controller.destroy();
+
+    assert.strictEqual(controller.get("assign_allowed_on_groups"), undefined);
```

Comment:
The absence of `if (this.isDestroying || this.isDestroyed)` will cause this to fail, obv.

Reply:
OHHH we can